File: sources/kotlin/09_fib_recursive.kt
// task 09 fib_recursive — expected output: 102334155
// build: kotlinc 09_fib_recursive.kt -include-runtime -d prog.jar    run: java -jar prog.jar    [native build: kotlinc-native -opt -o prog 09_fib_recursive.kt    native run: ./prog]

private fun fib(n: Long): Long {
    if (n < 2L) return n
    return fib(n - 1L) + fib(n - 2L)
}

fun main() {
    println(fib(40L))
}

File: sources/swift/09_fib_recursive.swift
// task 09 fib_recursive — expected output: 102334155
// build: swiftc -O -o prog 09_fib_recursive.swift    run: ./prog

func fib(_ n: Int64) -> Int64 {
    if n < 2 {
        return n
    }
    return fib(n - 1) + fib(n - 2)
}

print(fib(40))
